src/sentry_sync.h

```diff
@@ -258,7 +258,7 @@ static inline int
 sentry__atomic_fetch_and_add(volatile int *val, int diff)
 {
 #ifdef SENTRY_PLATFORM_WINDOWS
-    return InterlockedExchangeAdd(val, diff);
+    return (int)InterlockedExchangeAdd((volatile LONG *)val, (LONG)diff);
```

Comment:
Usually, `int` and `long` are both 32-bit, but generally I'm afraid that this might cause problems. If you're compiling this on a platform where `int` and `long` have different size, an overflow may write into memory after `int val`.

I'm actually glad that the linter caught that.

Reply:
windows 32-bit builds had a compiler warning here. TBH, I’m not quite sure what the correct thing to do here is.

Reply:
It seems that we only use this for refcounting, so you have two options:
1. Make it a `long` all the way. `__sync_fetch_and_add` also supports `long`.
2. Assert in the Windows case that `sizeof(int) == sizeof(long)`, then cast.